### PR TITLE
egnaddrs cli functionality

### PR DESCRIPTION
**Pull request by ricomateo**
**Description**
This PR adds the following changes
* `egnaddrs` CLI functionality used to output addresses for all relevant Eigenlayer and AVS contracts within the network.

### CODE REVIEW

File: crates/eigen-cli/src/eigen_address.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+use alloy_primitives::Address;
+use alloy_provider::Provider;
+use eigen_utils::{
+    binding::{DelegationManager, IBLSSignatureChecker, RegistryCoordinator},
+    get_provider,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct EigenAddresses {
```
Comment (dralves) [inline crates/eigen-cli/src/eigen_address.rs:13]: it's confusing to have an EigenAddresses and an EigenLayerAddresses stuct, could we rename that?

File: crates/eigen-cli/src/args.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    ArgGroup::new("manager_or_coordinator")
+        .required(true)
+        .args(&["service_manager", "registry_coordinator"]),
+))]
+#[command(
+    version,
+    about = "Used to help debug and test deployments and contract setups.",
+    long_about = "This utility facilitates the debugging and testing of Eigenlayer and AVS contract deployments by retrieving and displaying a comprehensive list of contract addresses. Starting from an initial contract address provided, it recursively identifies and prints addresses for all relevant Eigenlayer and AVS contracts within the network. This includes service managers, registry coordinators, and various registries, thus providing a view of the deployment's structure within the network."
+)]
+pub struct Args {
```
[the remark this answers is not in the document]
Comment (ricomateo) [reply crates/eigen-cli/src/args.rs:16]: Do you mean having all the clap stuff (for both `egnaddrs` and `egnkey`) in a single `args.rs` file?
We were thinking of having `egnaddrs` and `egnkey` as separated crates, each with its own `args.rs`.
Which way should we do it?

[the remark this answers is not in the document]
Comment (dralves) [reply crates/eigen-cli/src/args.rs:16]: yes, don't think there's an advantage in having two command lines, we might even have to add more subcommands in the future

[the remark this answers is not in the document]
Comment (ricomateo) [reply crates/eigen-cli/src/args.rs:10]: Done!

**Review by dralves — APPROVED** [review]

**Pull request merged**